### PR TITLE
Small refactors

### LINKED ISSUE #17: small refactor todos
This is a big list of todos that came up during the refactor of the Go module.

- [x] **Pluralize repeated fields in proto files**: Some of the proto structs have repeated fields which are not plural (eg `signature` vs `signatures`).
- [ ] **Audit use of the name "Outgoing"**: The old code used words like `OutgoingTx`, `OutgoingPool`, etc. to refer to the thing that is now known as `SendToEthereum`. We now use the word `OutgoingTx` to refer to something completely different. We need to carefully find old uses of this term (requires in depth knowledge of the system to make sure it isn't the new use of the term) and change them to `SendToEthereum`.
- [ ] **Audit use of "Create" vs "Build" vs "New"**: Both the Create and Build prefixes generally indicate a method that does something somewhat complicated and stateful to make a new entity of some kind. We should standardize on one or the other. "New" usually indicates a simpler instantiation of a struct, which does not mutate other state, but in some places it is doing something more and we should replace it with Create or Build.
- [ ] **Nonce prefixing**: We have many different nonces, including BatchNonce, SignerSetNonce, EventNonce, etc. Some data structures have two nonces. Mixing up nonces usually leads directly to a double spend error or something similarly dire, so we should prefix all nonce variables and fields with their type. Fixing this issue requires pretty in depth knowledge of what the different nonces are.
- [ ] **BridgeContract vs BridgeEthereumAddress**: Make sure this is consistent everywhere.
- [ ] **Deprecated slash fraction params**: We currently have different parameters for slashing of different types of OutgoingTxs. Unify them into one parameter (unless there is a need to actually slash different amounts).
- [x] **Audit all Validate and ValidateBasic functions, especially on Msgs**
- [ ] **Decide on, and likely remove MsgRequestBatch**: This message allows anyone to request that a batch of SendToEthereums for a given token be created. By spamming this endpoint, an adversary could cause a large number of batches with only a small number of transactions in each batch to be created, which would not be profitable to submit, and would effectively halt the bridge. We have now moved to automatic batch creation, so this message should no longer be necessary.
- [ ] **Double check that any denoms passed in are validated correctly**
- [x] **Validation for ContractCallTx**: ContractCallTxs are created by trusted external modules, calling a keeper method. We do not need to validate them for security reasons, but it might be nice to validate them for the convenience of calling module authors debugging. 
- [x] **Use bools in proto messages instead of magic values**: We have many proto request messages that either return an array of many results, or only the latest result. Currently, the caller indicates they want the latest result by passing in a magic value like 0 or "". This has already caused programming errors in the short time it's been around. We should switch to an explicit "latest" boolean in the request.
- [x] **Check if any iterators can be exited early**: There are many iterators in the codebase that can be exited early by returning `true`. There may be some that could be exited early, but are not, wasting iterations.

- [x] **Rename PendingSignerSetTxEthereumSignaturesRequest to UnsignedSignerSetTxsRequest**: This is much more readable and accurate.

### PR DESCRIPTION
**Pull request by cbrit**
Some minor items from an old issue. Issues not addressed will be covered by future work (slashing) or were already fixed and checked off.

I recommend looking at a commit at a time.

Closes #17

### CODE REVIEW

File: orchestrator/gorc/src/commands/cosmos_to_eth.rs [hunk 1]
```diff
@@ -60,121 +57,112 @@ impl Runnable for CosmosToEthCmd {
...
-                        panic!("Your transfer of {} {} tokens is greater than your balance of {} tokens. Remember you need some to pay for fees!", print_eth(amount.amount), gravity_denom, print_eth(found.amount.clone()));
-                    }
-                } else if amount.amount.clone() * times.into() >= found.amount {
-                    if is_cosmos_originated {
-                        panic!("Your transfer of {} * {} {} tokens is greater than your balance of {} tokens. Try to reduce the amount or the --times parameter", print_atom(amount.amount), times, gravity_denom, print_atom(found.amount.clone()));
-                    } else {
-                        panic!("Your transfer of {} * {} {} tokens is greater than your balance of {} tokens. Try to reduce the amount or the --times parameter", print_eth(amount.amount), times, gravity_denom, print_eth(found.amount.clone()));
+            println!("Cosmos balances {:?}", balances);
+
+            let times = self.args.get(4).expect("times is required");
```
Comment (EricBolten) [inline orchestrator/gorc/src/commands/cosmos_to_eth.rs:115]: I honestly think we should delete this `times` thing, it seems absolutely useless and just makes this more complicated.

File: module/proto/gravity/v1/msgs.proto [hunk 2]
```diff
@@ -19,9 +19,6 @@ service Msg {
       returns (MsgCancelSendToEthereumResponse) {
     // option (google.api.http).post = "/gravity/v1/send_to_ethereum/cancel";
   }
-  rpc RequestBatchTx(MsgRequestBatchTx) returns (MsgRequestBatchTxResponse) {
```
Comment (EricBolten) [inline module/proto/gravity/v1/msgs.proto:22]: Can we double check that the web UI isn't calling this?

**Review by EricBolten — COMMENTED** [review]
Some comments.

File: module/proto/gravity/v1/msgs.proto (hunk 2 above)
Comment (cbrit) [reply module/proto/gravity/v1/msgs.proto:22]: Confirmed with Elliot, who confirmed with Strangelove.

**Review by EricBolten — COMMENTED** [review]
One last suggested fix.

File: orchestrator/gorc/src/commands/cosmos_to_eth.rs [hunk 3]
```diff
@@ -112,56 +112,45 @@ impl Runnable for CosmosToEthCmd {
...
                 Some(found) => {
-                    if amount.amount.clone() * times.into() >= found.amount && times == 1 {
+                    if amount.amount.clone() >= found.amount {
                         if is_cosmos_originated {
                             panic!("Your transfer of {} {} tokens is greater than your balance of {} tokens. Remember you need some to pay for fees!", print_atom(amount.amount), gravity_denom, print_atom(found.amount.clone()));
                         } else {
                             panic!("Your transfer of {} {} tokens is greater than your balance of {} tokens. Remember you need some to pay for fees!", print_eth(amount.amount), gravity_denom, print_eth(found.amount.clone()));
                         }
-                    } else if amount.amount.clone() * times.into() >= found.amount {
+                    } else if amount.amount.clone() >= found.amount {
```
[the remark this answers is not in the document]
Comment (cbrit) [reply orchestrator/gorc/src/commands/cosmos_to_eth.rs]: Done

**Review by mvid — APPROVED** [review]

**Pull request merged**